### ISSUE TITLE
Spatial tests - 30D to 10D Presto performance comparison on crop type task

### ISSUE THREAD

**Issue by giollimirgia** [WorldCereal/presto-worldcereal#59]
Excluded `CROPTYPE_LABEL` : `[0, 991, 7900, 9900, 9998]` 
Distribution of training and test set after mapping to 9 classes (from 89 original classes): 

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/372cc2ca-74be-4727-8be8-65978ee5eceb alt="image" width="500" height="450">

Experiments:
- [x] pretrained Presto (not fine tuned) + Catboost training on crop-type monthly data
- [x] pretrained Presto (not fine tuned) + Catboost training on crop-type decadal data
- [x] Presto FT on (crop/no-crop) monthly data + Catboost training on crop-type monthly data
- [x] Presto FT on (crop/no-crop) decadal data + Catboost training on crop-type decadal data

**Comment by giollimirgia** [comment]
**pretrained Presto (not fine tuned) + Catboost training on crop-type monthly data**
<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/9bd81556-1de7-4cbb-9a8e-b0c9b042c73b alt="image" width="500" height="450">

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/7a8236e6-525b-4d4f-ba4f-35216ff4b338 alt="image" width="250" height="200">

**Comment by giollimirgia** [comment]
**pretrained Presto (not fine tuned) + Catboost training on crop-type decadal data**

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/94d3c7c7-f4e5-41f9-bd0d-e722a2f1bb88 alt="image" width="500" height="450">

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/3ff71932-7379-4525-8ade-00fa94248b3e alt="image" width="250" height="200">

**Comment by giollimirgia** [comment]
**Presto FT on (crop/no-crop) decadal data + Catboost training on crop-type decadal data**

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/aff12130-b973-4ae8-9abb-ad1dd469af65 alt="image" width="550" height="400">

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/76a63153-cecc-4f7b-84d5-d64e99824541 alt="image" width="250" height="200">

**Comment by giollimirgia** [comment]
**Presto FT on (crop/no-crop) monthly data + Catboost training on crop-type monthly data**

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/ae2238b4-8e72-4987-b44e-15e3e2ff455a alt="image" width="550" height="400">

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/3a411187-4535-453f-9d0e-f8ff55c12f80  alt="image" width="250" height="200">

**Comment by giollimirgia** [comment]
Additional experiments suggested by @cbutsko. the classes listed below might compromise the performance as the model cannot be validated in a fair way (eg model should not be penalized when predicting "wheat" when the label is "cereal").

Excluded `CROPTYPE_LABEL` : 
```
[ 0, 991, 7900, 9900, 9998, # unspecified cropland 
 1910, 1900, 1920, 1000, # cereals
 11, 9910, 6212  # tmp crops ]
```

Distribution of training, validation and test set after mapping to 9 classes (from 89 original classes): 
<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/1789a36f-af30-4348-bf7b-47047de1f907 alt="image" width="600" height="400">

Experiments:
- [x] pretrained Presto (not fine tuned) + Catboost training on crop-type monthly data
- [x] pretrained Presto (not fine tuned) + Catboost training on crop-type decadal data
- [x] Presto FT on (crop/no-crop) monthly data + Catboost training on crop-type monthly data
- [x] Presto FT on (crop/no-crop) decadal data + Catboost training on crop-type decadal data

**Comment by giollimirgia** [comment]
pretrained Presto (not fine tuned) + Catboost training on crop-type monthly data

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/ace50088-45fe-43ce-ac06-6afb840b93a5 alt="image" width="550" height="400">

```
             precision    recall  f1-score   support

   110101000       0.48      0.47      0.47      9804
   110102000       0.52      0.49      0.50     15380
   110103000       0.22      0.05      0.09      1466
   110106000       0.71      0.62      0.66      7800
   110107000       0.04      0.01      0.02       291
   110600001       0.70      0.84      0.77     13893
   110600002       0.41      0.05      0.09      1473
   110600003       0.15      0.09      0.12       275
   119999999       0.37      0.47      0.42      8063

    accuracy                           0.56     58445
   macro avg       0.40      0.34      0.35     58445
weighted avg       0.55      0.56      0.54     58445

 f1_score: 0.34699248297096724
```

**Comment by giollimirgia** [comment]
**Pretrained Presto (not fine tuned) + Catboost training on crop-type decadal data**

```
              precision    recall  f1-score   support

   110101000       0.42      0.58      0.48      9804
   110102000       0.53      0.38      0.45     15380
   110103000       0.11      0.03      0.05      1466
   110106000       0.55      0.68      0.61      7800
   110107000       0.05      0.01      0.02       291
   110600001       0.75      0.81      0.78     13893
   110600002       0.61      0.18      0.28      1473
   110600003       0.11      0.16      0.13       275
   119999999       0.41      0.39      0.40      8063

    accuracy                           0.54     58445
   macro avg       0.39      0.36      0.36     58445
weighted avg       0.54      0.54      0.53     58445

 f1_score: 0.3561279859356788
```
<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/654c2975-3132-4657-9489-1b36fcea1354 alt="image" width="550" height="400">

**Comment by giollimirgia** [comment]
**Presto FT on (crop/no-crop) monthly data + Catboost training on crop-type monthly data**

```
              precision    recall  f1-score   support

   110101000       0.52      0.44      0.48      9804
   110102000       0.55      0.59      0.57     15380
   110103000       0.23      0.17      0.19      1466
   110106000       0.74      0.72      0.73      7800
   110107000       0.00      0.00      0.00       291
   110600001       0.79      0.85      0.82     13893
   110600002       0.64      0.27      0.38      1473
   110600003       0.11      0.15      0.12       275
   119999999       0.42      0.48      0.45      8063

    accuracy                           0.60     58445
   macro avg       0.45      0.41      0.42     58445
weighted avg       0.60      0.60      0.60     58445

 f1_score: 0.41539206463757516
```
<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/c3dbd105-8243-4e92-9dd5-b3c2301f68c3 alt="image" width="500" height="400">

**Comment by giollimirgia** [comment]
**Presto FT on (crop/no-crop) decadal data + Catboost training on crop-type decadal data**

```
              precision    recall  f1-score   support

   110101000       0.45      0.52      0.48      9804
   110102000       0.56      0.54      0.55     15380
   110103000       0.39      0.10      0.16      1466
   110106000       0.64      0.72      0.68      7800
   110107000       0.01      0.00      0.00       291
   110600001       0.86      0.82      0.84     13893
   110600002       0.63      0.14      0.23      1473
   110600003       0.13      0.13      0.13       275
   119999999       0.41      0.48      0.44      8063

    accuracy                           0.59     58445
   macro avg       0.45      0.38      0.39     58445
weighted avg       0.60      0.59      0.59     58445

 f1_score: 0.38961371241542064
```

<img src=https://github.com/WorldCereal/presto-worldcereal/assets/49987400/1da58bdc-a09d-4ec4-b95b-3bd12c7edf5f
 alt="image" width="500" height="400">

**Comment by giollimirgia** [comment]
Experiment F1 score summary

![image](https://github.com/WorldCereal/presto-worldcereal/assets/49987400/236862eb-4ed8-4593-b4ac-a9613936d089)

**Comment by giollimirgia** [comment]
Experiment F1 score summary

![image](https://github.com/WorldCereal/presto-worldcereal/assets/49987400/ec448a13-30a8-451a-8524-15a783da17ce)

**Comment by giollimirgia** [comment]
Experiment F1 score summary

![image](https://github.com/WorldCereal/presto-worldcereal/assets/49987400/dce2ff45-1d17-41d0-9e1a-90856f88ed5b)

**Comment by gabrieltseng** [comment]
Closing - summary in #63
